test/controllers/sessions_controller_test.rb

```diff
@@ -0,0 +1,28 @@
+require 'test_helper'
+
+class SessionsControllerTest < ActionController::TestCase
+
+  def setup
+    request.env["omniauth.auth"] = authenticate_with_github(users(:angela))
+  end
+
+  test "should login" do
+    assert_no_difference 'User.count' do
+      get :create, provider: 'github'
+    end
+    assert_difference('User.count', 1) do
+      request.env["omniauth.auth"] = authenticate_with_github(User.new(username: 'linatodi', token: SecureRandom.hex(8), email: 'torresdiazlina@gmail.com'))
+      get :create, provider: 'github'
+    end
+    assert_redirected_to gems_path
+    assert_equal 'Signed in!', flash[:success]
+  end
```

Comment:
Can you split this test into 2. It's a good idea to give a test only one reason to fails and I think here you are giving 2 reasons by checking the behaviour when a user is new and when its not. The clearer a test is the more value they add.
It would look something like this:
```ruby
test "should login with an existing user" do
  existing_user = users(:angela)
  authenticate_with_github(existing_user)
  assert_no_difference 'User.count' do
    get :create, provider: 'github', 
  end
  assert_redirected_to gems_path
end

test "should login with a new user" do
  new_user = User.new(username: 'linatodi', token: SecureRandom.hex(8), email: 'torresdiazlina@gmail.com')
  authenticate_with_github(new_user)
  assert_difference('User.count', 1) do
    get :create, provider: 'github'
  end
  assert_redirected_to gems_path
end
```